src/test/java/uk/gov/pay/directdebit/payments/services/SandboxServiceTest.java

```diff
@@ -1,87 +1,100 @@
+        verify(mockedPayerService).createOrUpdatePayer(mandateFixture.getExternalId(), gatewayAccountFixture.toEntity(), createPayerRequest);
     }
 
     @Test
-    public void confirm_shouldRegisterAPaymentSubmittedToProviderEventWhenSuccessfullyConfirmed() {
+    public void confirm_shouldRegisterAPaymentSubmittedToProviderEventWhenSuccessfullyConfirmingOneOffMandate() {
+        MandateFixture mandateFixture = aMandateFixture().withMandateType(MandateType.ONE_OFF).withGatewayAccountFixture(gatewayAccountFixture);
+        TransactionFixture transactionFixture = TransactionFixture.aTransactionFixture().withMandateFixture(mandateFixture);
         ConfirmationDetails confirmationDetails = confirmationDetails()
-                .withTransaction(aTransactionFixture())
-                .withMandate(aMandateFixture())
+                .withMandateFixture(mandateFixture)
+                .withTransactionFixture(transactionFixture)
                 .build();
-        Transaction transaction = confirmationDetails.getTransaction();
+        Map<String, String> details = ImmutableMap.of("sort_code", "123456", "account_number", "12345678", "transaction_external_id", transactionFixture.getExternalId());
+        when(mockedMandateConfirmService
+                .confirm(mandateFixture.getExternalId(), details))
+                .thenReturn(confirmationDetails);
 
+        service.confirm(mandateFixture.getExternalId(), gatewayAccountFixture.toEntity(), details);
+        verify(mockedTransactionService).paymentSubmittedToProviderFor(transactionFixture.toEntity(), LocalDate.now().plusDays(4));
+    }
+
+    @Test
+    public void confirm_shouldNotRegisterAPaymentSubmittedToProviderEventWhenSuccessfullyConfirmingOnDemandMandate() {
+        MandateFixture mandateFixture = aMandateFixture().withMandateType(MandateType.ON_DEMAND).withGatewayAccountFixture(gatewayAccountFixture);
+        ConfirmationDetails confirmationDetails = confirmationDetails()
+                .withMandateFixture(mandateFixture)
+                .build();
         Map<String, String> details = ImmutableMap.of("sort_code", "123456", "account_number", "12345678");
-        when(mockedPaymentConfirmService.confirm(gatewayAccount.getExternalId(), paymentRequestExternalId, details))
+        when(mockedMandateConfirmService
+                .confirm(mandateFixture.getExternalId(), details))
                 .thenReturn(confirmationDetails);
-        when(mockedPayerService.getPayerFor(transaction)).thenReturn(payer);
 
-        service.confirm(paymentRequestExternalId, gatewayAccount, details);
-        verify(mockedTransactionService).paymentSubmittedToProviderFor(transaction, payer, confirmationDetails.getMandate(), LocalDate.now().plusDays(4));
+        service.confirm(mandateFixture.getExternalId(), gatewayAccountFixture.toEntity(), details);
+        verifyNoMoreInteractions(mockedTransactionService);
```

Comment:
Since there aren’t any verifications at all, I think `verifyZeroInteractions` would be clearer here